src/Renci.SshNet/Abstractions/StreamExtensions.cs

```diff
@@ -8,8 +9,15 @@ internal static class StreamExtensions
     {
         public static ValueTask DisposeAsync(this Stream stream)
         {
-            stream.Dispose();
-            return default;
+            try
+            {
+                stream.Dispose();
+                return default;
+            }
+            catch (Exception exc)
+            {
+                return new ValueTask(Task.FromException(exc));
+            }
```

Comment:
unrelated, just copying the netcore impl

https://github.com/dotnet/runtime/blob/1d1bf92fcf43aa6981804dc53c5174445069c9e4/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L180C17-L187C10